test/extensions/filters/http/transformation/inja_transformer_test.cc

```diff
@@ -936,6 +936,120 @@ TEST(Transformer, transformHeaderAndHeadersToRemove) {
   EXPECT_EQ("custom-value3", result3[0]->value().getStringView());
 }
 
+TEST(InjaTransformer, ReplaceWithRandomBodyTest) {
+  Http::TestRequestHeaderMapImpl headers{{":method", "GET"}, {":path", "/foo"}};
+  TransformationTemplate transformation;
+
+  auto pattern = "replace-me";
+  auto test_string = "test-replace-me";
+  auto formatted_string = fmt::format("{{{{replace_with_random(\"{}\", \"{}\")}}}}", test_string, pattern);
+
+  transformation.mutable_body()->set_text(formatted_string);
+
+  InjaTransformer transformer(transformation);
+
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks;
+
+  Buffer::OwnedImpl body("");
+  transformer.transform(headers, &headers, body, callbacks);
```

Comment:
shouldn't the body here be nonempty and contain `test_string`? i must be missing something because the test presumably somehow passes and replaces the replace-me